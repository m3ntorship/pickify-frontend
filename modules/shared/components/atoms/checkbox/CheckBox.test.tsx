import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Checkbox from './index';

describe('Snapshot tests for Checkbox component', () => {
  it('Testing Checkbox size Default and not disabled', () => {
    const tree = renderer
      .create(<Checkbox size="Default" disabled={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing Checkbox size Small and not disabled', () => {
    const tree = renderer
      .create(<Checkbox size="Small" disabled={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing Checkbox size Default and disabled true', () => {
    const tree = renderer.create(<Checkbox size="Default" disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing Checkbox size Small and  disabled true', () => {
    const tree = renderer.create(<Checkbox size="Small" disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Behavioral tests for Checkbox', () => {
  it("check if it's not clicked/checked as default", () => {
    render(<Checkbox size="Default" disabled={false} />);
    expect(screen.getByRole('checkbox')).not.toBeChecked();
  });

  it('Checkbox not clicked while it is disabled', () => {
    const mockClick = jest.fn();
    render(<Checkbox size="Default" disabled onMockClick={mockClick} />);
    const input = screen.getByTestId('input-test');
    userEvent.click(input);
    const zero = 0;
    expect(mockClick).toHaveBeenCalledTimes(zero);
  });
});
