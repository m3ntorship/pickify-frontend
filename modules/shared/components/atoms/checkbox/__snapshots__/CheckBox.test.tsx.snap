// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for Checkbox component Testing Checkbox size Default and disabled true 1`] = `
<label
  className="label size-default"
  htmlFor="checkbox"
>
  <input
    checked={false}
    className="input-unchecked input-unchecked-disabled size-default"
    data-testid="input-test"
    disabled={true}
    id="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
</label>
`;

exports[`Snapshot tests for Checkbox component Testing Checkbox size Default and not disabled 1`] = `
<label
  className="label label-undisabled size-default"
  htmlFor="checkbox"
>
  <input
    checked={false}
    className="input-unchecked input-unchecked-undisabled size-default"
    data-testid="input-test"
    disabled={false}
    id="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
</label>
`;

exports[`Snapshot tests for Checkbox component Testing Checkbox size Small and  disabled true 1`] = `
<label
  className="label size-small"
  htmlFor="checkbox"
>
  <input
    checked={false}
    className="input-unchecked input-unchecked-disabled size-small"
    data-testid="input-test"
    disabled={true}
    id="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
</label>
`;

exports[`Snapshot tests for Checkbox component Testing Checkbox size Small and not disabled 1`] = `
<label
  className="label label-undisabled size-small"
  htmlFor="checkbox"
>
  <input
    checked={false}
    className="input-unchecked input-unchecked-undisabled size-small"
    data-testid="input-test"
    disabled={false}
    id="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
</label>
`;
